client/my-sites/site-settings/podcasting-details/index.jsx

```diff
@@ -195,12 +225,33 @@ const getFormSettings = settings => {
 	] );
 };
 
-const connectComponent = connect( state => {
+const connectComponent = connect( ( state, ownProps ) => {
+	const siteId = getSelectedSiteId( state );
+
+	let podcastingCategoryId = null;
+
+	// Retrieve podcasting category ID from saved category slug (if any)
+	const podcastingCategorySlug = get( ownProps.fields, 'podcasting_archive' );
+	if ( podcastingCategorySlug && podcastingCategorySlug !== '0' ) {
+		const categories = getTerms( state, siteId, 'category' );
+		if ( categories && categories.length ) {
+			// This is not a bound selector
+			// eslint-disable-next-line wpcalypso/redux-no-bound-selectors
+			const podcastingCategory = find( categories, c => c.slug === podcastingCategorySlug );
+			if ( podcastingCategory ) {
+				podcastingCategoryId = podcastingCategory.ID;
+			}
+		}
+	}
+
 	return {
+		siteId,
 		siteSlug: getSelectedSiteSlug( state ),
+		podcastingCategoryId,
+		isRequestingCategories: isRequestingTermsForQueryIgnoringPage( state, siteId, 'category', {} ),
 	};
 } );
 
-export default flowRight( connectComponent, wrapSettingsForm( getFormSettings ) )(
-	localize( PodcastDetails )
+export default flowRight( wrapSettingsForm( getFormSettings ), connectComponent )(
```

Comment:
I think `connect` is a decent place to put the logic that maps a category slug to a category ID.  In order to make this work, `wrapSettingsForm` needs to happen before `connectComponent`.